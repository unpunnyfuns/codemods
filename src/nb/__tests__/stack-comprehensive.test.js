import { describe, expect, it } from 'vitest'
import { testTransform } from '../../__tests__/test-helper.js'

describe('nb/stack-comprehensive', () => {
  it('handles all edge cases: multi-property expansion, tokens, flexbox, position', () => {
    const output = testTransform('nb/stack', 'nb/stack-comprehensive', 'tsx')
    expect(output).toMatchSnapshot()
  })
})
