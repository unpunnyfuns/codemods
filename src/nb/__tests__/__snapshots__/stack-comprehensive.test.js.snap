// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`nb/stack-comprehensive > handles all edge cases: multi-property expansion, tokens, flexbox, position 1`] = `
"import { SomeOtherComponent } from '@org/common/src/other'
import { Stack } from '@hb-frontend/app/src/components/nordlys/';
import { color, radius } from '@hb-frontend/nordlys';
import { StyleSheet } from 'react-native';

export function ComprehensiveExample() {
  return (
    <>
      {/* Test multi-property expansion with tokens */}
      <Stack direction='vertical' gap="md" style={styles.vstack0}>
        <SomeOtherComponent />
      </Stack>
      {/* Test marginX/marginY expansion */}
      <Stack direction='horizontal' gap="sm" style={styles.hstack0}>
        <SomeOtherComponent />
      </Stack>
      {/* Test flexbox props */}
      <Stack direction='vertical' style={styles.vstack1}>
        <SomeOtherComponent />
      </Stack>
      {/* Test position props */}
      <Stack direction='horizontal' style={styles.hstack1}>
        <SomeOtherComponent />
      </Stack>
      {/* Test full prop names */}
      <Stack direction='vertical' style={styles.vstack2}>
        <SomeOtherComponent />
      </Stack>
      {/* Test mixed shortcuts and value mappings */}
      <Stack direction='horizontal' style={styles.hstack2}>
        <SomeOtherComponent />
      </Stack>
      {/* Test edge case: boxSize instead of size */}
      <Stack direction='vertical' style={styles.vstack3}>
        <SomeOtherComponent />
      </Stack>
      {/* Test all border radius corners */}
      <Stack direction='horizontal' style={styles.hstack3}>
        <SomeOtherComponent />
      </Stack>
    </>
  );
}

const styles = StyleSheet.create({
  hstack0: {
    marginLeft: 4,
    marginRight: 4,
    marginTop: 2,
    marginBottom: 2
  },

  hstack1: {
    position: "absolute",
    top: 10,
    left: 20,
    zIndex: 999
  },

  hstack2: {
    backgroundColor: color.red['500'],
    padding: 3,
    marginHorizontal: 2,
    alignItems: 'center',
    justifyContent: 'space-between',
    flex: 1,
    overflow: "hidden",
    display: "flex"
  },

  hstack3: {
    borderTopLeftRadius: radius.xs,
    borderTopRightRadius: radius.sm,
    borderBottomLeftRadius: radius.md,
    borderBottomRightRadius: radius.lg
  },

  vstack0: {
    borderTopLeftRadius: radius.lg,
    borderTopRightRadius: radius.lg,
    borderBottomLeftRadius: radius.md,
    borderBottomRightRadius: radius.md,
    width: "full",
    height: "full"
  },

  vstack1: {
    flex: 1,
    flexGrow: 2,
    flexShrink: 0,
    flexWrap: "wrap",
    alignSelf: "center"
  },

  vstack2: {
    margin: 8,
    padding: 4,
    width: "100%",
    height: 200,
    backgroundColor: color.blue['500'],
    borderRadius: radius.xl,
    borderWidth: 2,
    borderColor: "gray.200"
  },

  vstack3: {
    width: 100,
    height: 100,
    borderTopLeftRadius: radius.sm,
    borderBottomLeftRadius: radius.sm,
    borderTopRightRadius: radius.md,
    borderBottomRightRadius: radius.md
  }
});
"
`;
